py-shared/quilt_shared/pkgpush.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+import enum
+import typing as T
+
+import pydantic
+
+from .aws import AWSCredentials
+from .types import NonEmptyStr
+
+if T.TYPE_CHECKING:
+    from types_aiobotocore_s3.type_defs import CopySourceTypeDef
+
+    from quilt3.util import PhysicalKey
+
+
+class TopHash(pydantic.ConstrainedStr):
+    min_length = 64
+    max_length = 64
+    regex = r"^[0-9a-f]+$"
+    strip_whitespace = True
+    to_lower = True
+
+
+class S3ObjectSource(pydantic.BaseModel):
```

Comment:
Maybe move this and other types to aws?

Reply:
maybe, tho now they only make sense in this context